### PR TITLE
[#3948] improvement(client-python): Add the simple auth provider in Python

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

Support for authentication framework and Simple Auth in Python Client.

### Why are the changes needed?

Fix: #3948 

### How was this patch tested?

Add the UTs and ITs.

### CODE REVIEW

**Comment by jerryshao** [comment]
@noidname01 would you please also help to review?

**Comment by jerryshao** [comment]
LGTM.

File: clients/client-python/gravitino/auth/auth_data_provider.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"""
+Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2.
+"""
+
+from abc import ABC, abstractmethod
+
+
+class AuthDataProvider(ABC):
```
Comment (noidname01) [inline clients/client-python/gravitino/auth/auth_data_provider.py:9]: It seems that this class has extended `Closeable` interface in Java and `close()` method would be used in other kinds of auth provider, so I think it is better to also support it here.

**Comment by noidname01** [comment]
https://github.com/datastrato/gravitino/blob/f958335814a96e649c43d36b8d2bf3065b8dfa9d/clients/client-python/gravitino/client/gravitino_admin_client.py#L30

This comment should also be removed.

**Comment by xloya** [comment]
@noidname01 Do you have any more comments on this?

**Comment by noidname01** [comment]
@xloya No, LGTM👍

**Comment by noidname01** [comment]
@xloya Oh wait, I just run `black` in this PR branch and found there is a file should be formatted, can you check this out?

**Comment by xloya** [comment]
> @xloya Oh wait, I just run `black` in this PR branch and found there is a file should be formatted, can you check this out?

Have fixed.

**Comment by jerryshao** [comment]
@noidname01 do you have further comments?

**Comment by noidname01** [comment]
> @noidname01 do you have further comments?

No, LGTM👍

**Review by noidname01 — APPROVED** [review]

**Review by jerryshao — APPROVED** [review]

**Pull request merged**